public/app/features/org/SelectOrgPage.tsx

```diff
@@ -19,17 +21,23 @@ const navModel = {
 const getUserOrgs = async () => {
   return await getBackendSrv().get('/api/user/orgs');
 };
-const setUserOrg = async (org: UserOrg) => {
-  return await getBackendSrv()
-    .post('/api/user/using/' + org.orgId)
-    .then(() => {
-      window.location.href = config.appSubUrl + '/';
-    });
+
+const mapDispatchToProps = {
+  setUserOrganization,
 };
 
-export const SelectOrgPage: FC = () => {
+const connector = connect(null, mapDispatchToProps);
+
+type Props = ConnectedProps<typeof connector>;
+
+export const SelectOrgPage: FC<Props> = ({ setUserOrganization }) => {
   const [orgs, setOrgs] = useState<UserOrg[]>();
 
+  const setUserOrg = async (org: UserOrg) => {
+    setUserOrganization(org.orgId);
```

Comment:
This action does an async API request, but looking at the code, the page location will be changed possibly before the request is completed. One issue I noticed is, if for some reason changing org fails, the user will still be redirected to the root page, but for the wrong org. Also I think it's a bit inconsistent that we set location this time directly in component, but other time in redux action.

Reply:
Good call @Clarity-89 my morning brain totally skipped that! @thisisobate I think we should take another look at this.

Solve the async issue @Clarity-89 mentioned and rethink how we want to do the redirection. IMO I'd keep the action clean and do the redirection in the components.

Reply:
Good catch! @Clarity-89 would open a separate PR for this. @natellium I agree with you...makes sense for redirections to happen in the components